### ISSUE TITLE
installation issue?

### ISSUE THREAD

**Issue by rhaynes74** [oniony/TMSU#220]
Hi folks, I installed tmsu on my mac but downloading the binary and copying it to /usr/local/bin (my mac won't let me copy to /usr/bin even as sudo.  Put the zsh completions in /usr/local/share/...   running tmsu init gives:

tmsu init
zsh: exec format error: tmsu

Any suggestions?

**Comment by oniony** [comment]
I don't have a Mac binary, so this would suggest you're trying to run an Linux binary?

You'll need to compile from source code.

See also https://github.com/oniony/TMSU/issues/186

**Comment by rhaynes74** [comment]
Yes, thanks.  I got it to work, there were, however, several modifications required to the makefile on the mac.   In particular, the -t flag in the cp commands don’t seem to work, also the -perm /+x in the runall script gave problems, and finally on the mac even sudo can’t seem to write to /usr/bin or /usr/share, I had to change these to /usr/local/bin and /usr/local/share.

Sincerely,

Dr. Ronald D. Haynes

Professor, Department of Mathematics and Statistics
Chair, MSc and PhD Scientific Computing Programs
Memorial University of Newfoundland

We acknowledge that the lands on which Memorial University’s campuses are situated are in the traditional territories of diverse Indigenous groups, and we acknowledge with respect the diverse histories and cultures of the Beothuk, Mi’kmaq, Innu, and Inuit of this province.

**Comment by dferlemann** [comment]
> Yes, thanks.  I got it to work, there were, however, several modifications required to the makefile on the mac.   In particular, the -t flag in the cp commands don’t seem to work, also the -perm /+x in the runall script gave problems, and finally on the mac even sudo can’t seem to write to /usr/bin or /usr/share, I had to change these to /usr/local/bin and /usr/local/share. Sincerely, Dr. Ronald D. Haynes Professor, Department of Mathematics and Statistics Chair, MSc and PhD Scientific Computing Programs Memorial University of Newfoundland We acknowledge that the lands on which Memorial University’s campuses are situated are in the traditional territories of diverse Indigenous groups, and we acknowledge with respect the diverse histories and cultures of the Beothuk, Mi’kmaq, Innu, and Inuit of this province.
> […](#)
> On Nov 20, 2020, 4:19 AM -0330, Paul Ruane ***@***.***>, wrote: I don't have a Mac binary, so this would suggest you're trying to run an Linux binary? You'll need to compile from source code. See also #186 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

I removed -t flag, - perm /+x in test/runall script, added local to all paths in the makefile.  I can confirm it works on MacOS Big Sur.
